src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -509,18 +513,22 @@ constexpr DataBox<tmpl::list<Tags...>>::DataBox(
     tmpl::list<AddMutableItemTags...> /*meta*/,
     tmpl::list<AddImmutableItemTags...> /*meta*/, Args&&... args) noexcept {
 #ifdef SPECTRE_DEBUG
-  expand_pack(detail::check_initialization_argument_type<
-              AddMutableItemTags, typename AddMutableItemTags::type,
-              std::decay_t<Args>>()...);
+  if constexpr (sizeof...(Args) > 0) {
+    expand_pack(detail::check_initialization_argument_type<
+                AddMutableItemTags, typename AddMutableItemTags::type,
+                std::decay_t<Args>>()...);
+  }
 #endif  // SPECTRE_DEBUG
 
   merge_old_box(std::forward<Box>(old_box), KeepTagsList{});
 
-  std::tuple<Args&&...> args_tuple(std::forward<Args>(args)...);
+  if constexpr (sizeof...(Args) > 0) {
+    std::tuple<Args&&...> args_tuple(std::forward<Args>(args)...);
 
-  add_items_to_box(args_tuple, tmpl::list<AddMutableItemTags...>{},
-                   std::make_index_sequence<sizeof...(AddMutableItemTags)>{},
-                   tmpl::list<AddImmutableItemTags...>{});
+    add_items_to_box(args_tuple, tmpl::list<AddMutableItemTags...>{},
```

Comment:
again I think you want to call the checks on the immutable_tags if `size_of..(Args)==0`